articles/choosing-between-firebase-and-mongodb/index.md

```diff
@@ -0,0 +1,95 @@
+### Features of MongoDB
+1. Native Replication.
+MongoDB provides a replication feature by distributing data across different machines through a replica set topology.According to [Congruent Solutions](https://www.congruentsolutions.com/blogposts/mongodb-replication-approach-setup-using-arbiter/),A replica set is a group of MongoDB instances that host the same data set.In a replica set one node is primary node that receives all write operations and all other nodes acts as secondaries. Replica set can have only one primary node.After each successive writes to the primary node, MongoDB will replicate the same data to all the secondary nodes present in the replica set.
+2. Supports Ad-hoc Queries.
+MongoDB supports ad-hoc queries by indexing the Binary JSON ( BSON) documents.
+3. It is schemaless.
+MongoDB is considered flexible since its database is written in the C++ programming language.
+4. High Performance.
+MongoDB's performance is high as compared to other databases due to its scalability, and it has a faster query response because of replication and indexing thus making it a better choice for real-time applications.
+5. Indexing.
+MongoDB indexes improve search performance.
+### Advantages of MongoDB
+1. It contains heterogeneous data.
+2. It provides high performance, scalability, and availability.
+3. It can be integrated with Big Data Hadoop.
+4. MongoDB's documentation has an extensive collection of literature and MongoDB tutorials for new users.
+5. MongoDB has high data security since there is no risk of SQL injection.
+### Disadvantages of MongoDB
+1. There is high data consumption because of denormalization.
+2. It has no joins. We need to make many queries and join data.
+3. MongoDB has a limited data size.
+4. Indexing and searching in MongoDB is not powerful.
+### History of Firebase
+Firebase is a cloud service provider and backend as a service company based in San Francisco, California, founded in September 2011 by James Tamplin and Andrew Lee as Envolve. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website.
+### What is firebase
+According to [Cypress](https://www.cypressoft.com/post/choose-firebase-over-backend-development/),Firebase is a toolkit and infrastructure that aims at supporting the process of building better applications and websites.
+Firebase is a Back-end as a service owned by Google which provides a server-less backend to app and web developers. Firebase offers a wide range of backend functionalities which include authentication,real-time database, storage, analytics, crash reporting, and dynamic links.
```

Comment:
First sentence -> Already mentioned above

Keep second sentence.

Unclear whether those features are also in MongoDB or not